### PR TITLE
Allow firewall matching using route parameters

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #31362 
| License       | MIT
| Doc PR        | no doc PR yet

Implementation of https://github.com/symfony/symfony/issues/31362#issuecomment-489312472

### CODE REVIEW

**Comment by MatTheCat** [comment]
Uh https://travis-ci.org/symfony/symfony/jobs/531801914 fails because `assertResponseStatusCodeSame` doesn not exist. Am I supposed to not use it?

**Comment by linaori** [comment]
I can see the reason as of why this configuration method can be preferred over url matching. I'm a bit afraid that this list might end up rather big and that the `security.yaml` will have the majority of lines containing this particular config, making it harder to maintain.

Would it be an idea to add pattern matching for route names? For example: `app.foo.*` would match everything starting with `app.foo.`. This is a crude example and might be a bit difficult to properly realize with regexes because of the `.` in names being allowed.

**Comment by MatTheCat** [comment]
I was afraid of that. Plus changing a route name would mean changing it in these lists too. Are you sure we cannot have the firewall configured in the route instead? :smile:

**Comment by nicolas-grekas** [comment]
What about matching a route parameter instead of its name?

**Comment by MatTheCat** [comment]
@nicolas-grekas sounds like my first proposal; WDYT of https://github.com/symfony/symfony/issues/31362#issuecomment-488674941 ?

**Comment by MatTheCat** [comment]
Okay so I implemented https://github.com/symfony/symfony/pull/31485#issuecomment-494086813 but AppVeyor won’t tell me why it fails.

File: src/Symfony/Bundle/SecurityBundle/RouteParamsMatcher.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
```
Comment (ismail1432) [inline src/Symfony/Bundle/SecurityBundle/RouteParamsMatcher.php]: miss one *

Comment (MatTheCat) [reply src/Symfony/Bundle/SecurityBundle/RouteParamsMatcher.php]: I copied this part from another file so I don't think it's wrong.

Comment (ismail1432) [reply src/Symfony/Bundle/SecurityBundle/RouteParamsMatcher.php]: You’re right sorry 👍🏽

**Pull request closed without merging**